libuuu/cmd.cpp

```diff
@@ -237,21 +238,17 @@ int get_string_in_square_brackets(string &cmd, string &context)
 uint32_t str_to_uint(string &str)
 {
 	uint32_t val = 0;
-	const char * start = str.substr(0).c_str();
-	char * endPtr = const_cast<char*>(start);
+        std::stringstream ss;
 	if (str.size() > 2 && 
 	    str.substr(0, 2).compare("0x") == 0)
 	{
-		val =  strtoul(start+2, &endPtr, 16);
+            ss << std::hex << str.substr(2);
 	}
```

Comment:
what's problem do you try to fix? the function should be the same

Reply:
don't know why but the old code fail to convert on x86_64 latest ubuntu.... and well c++ conversion function are more robust than the strtoul legacy

Reply:
It pass ubuntu 18.08 at auto build system.